src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -24,8 +24,19 @@ internal sealed class EnumConverter<T> : JsonConverter<T>
 
         private readonly JsonNamingPolicy? _namingPolicy;
 
-        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCache;
-
+        /// <summary>
+        /// Holds a mapping from enum value to text that might be formatted with <see cref="_namingPolicy" />.
+        /// </summary>
+        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _namingPolicyCache;
```

Comment:
This predates your own changes, but do you know why we're keying the caches on `ulong`? This could easily just be `T` and that wouldn't have any perf impact since enums and their equality is just represented as integers at runtime.

Reply:
Earlier measurements showed private bytes savings when a singlular type (`ulong`) was used as the key - https://github.com/dotnet/runtime/pull/36726#discussion_r428868336.

Reply:
Sounds good.